packages/service-worker/worker/test/happy_spec.ts

```
@@ -1230,6 +1246,36 @@ import {SwTestHarness, SwTestHarnessBuilder} from '../testing/scope';
         server.assertSawRequestFor('/foo.txt');
       });
 
+      it('keeps serving api requests with freshness strategy when failing to write to cache',
+         async() => {
+           // Initialize the SW.
+           expect(await makeRequest(scope, '/foo.txt')).toEqual('this is foo');
+           await driver.initialized;
+           server.clearRequests();
+
+           // Make the caches unwritable.
+           spyOn(MockCache.prototype, 'put').and.throwError('Can\'t touch this');
+
+           expect(await makeRequest(scope, '/api/foo')).toEqual('this is api foo');
+           expect(driver.state).toBe(DriverReadyState.NORMAL);
+           server.assertSawRequestFor('/api/foo');
+         });
```

Could you also add an assertion that an error was logged with `DebugHandler` (here and below)?
(I know we usually don't do that in other tests, but I think it is a good idea, especially here that we basically ignore the error.)

---

Should I create `MockDebugHandler` or spying on existing is ok?

---

Implemented spying on existing `DebugHandler`. Feel free to start another conversation if anything is not ok.

---

Spying on the existing one sounds good. Thx!